qemu/tests/ksm_ksmtuned.py

```diff
@@ -0,0 +1,103 @@
+import re
+import os
+import time
+import logging
+
+from shutil import copyfile
+
+from avocado.utils import process
+
+from virttest import arch
+from virttest import env_process
+from virttest import utils_misc
+from virttest.utils_test import VMStress
+from virttest.staging import utils_memory
+
+
+def run(test, params, env):
+    """
+    Check KSM can be started automaticly when ksmtuned threshold is reached
+
+    1. Get the memory of your host and the KSM_THRES_COEF
+    2. Boot a guest with memory less than KSM_THRES_COEF threshold
+    3. Get the memory used in host of process qemu-kvm
+    4. Get the free memory in host
+    5. If both the free memory size is not smaller than the threshold and guest
+        used memory + threshold is not bigger than total memory in host. Check
+        the ksm status in host. Ksm should not start in the host
+    6. Repeat step 2~5 under it broke the rule in step 5
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def check_ksm(mem, stress=False):
+        """
+        :param mem: Boot guest with given memory, in KB
+        :param stress: Load stress or not
+        """
+        params['mem'] = mem // 1024
+        params['start_vm'] = 'yes'
+        vm_name = params['main_vm']
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        vm.wait_for_login()
+        if stress:
+            params['stress_args'] = ('--cpu 4 --io 4 --vm 2 --vm-bytes %sM' %
+                                     (int(params['mem']) // 2))
+            stress_test = VMStress(vm, "stress", params)
+            stress_test.load_stress_tool()
+            time.sleep(30)
+        qemu_pid = vm.get_pid()
+        qemu_used_page = utils_misc.normalize_data_size(process.getoutput(
+            params['cmd_get_qemu_used_mem'] % qemu_pid, shell=True) + 'K', 'B')
+        qemu_used_mem = int(float(qemu_used_page)) * 4
```

Comment:
4K is for x86 but for ppc, it should be 64k
[root@ibm-p9b-19 kar]# getconf PAGE_SIZE
65536

Reply:
@MiriamDeng Thanks a lot. Updated, please help re-review.